webextension/background_scripts/WEPersonaSwitcher.js

```diff
@@ -0,0 +1,84 @@
+// Verify if we need to load the default preferences by checking if the 
+// default_loaded flag is undefined. 
+var checkDefaultsLoaded = browser.storage.local.get("defaults_loaded");
+checkDefaultsLoaded.then(result => { if(undefined === result.defaults_loaded) {
+										loadDefaults();
+									}else{
+										onError(result.error);
+									}});
+
+function loadDefaults(){
+	var setting = browser.storage.local.set(
+		{
+			default_loaded: true,
```

Comment:
Checking against preference defaults_loaded, but setting preference default_loaded. Assuming this is meant to be the same preference?